src/operations/blas2/gemv.hpp

```diff
@@ -921,5 +921,85 @@ SYCL_BLAS_INLINE void GemvCol<Lower, Diag, Upper, Unit, lhs_t, matrix_t,
   vector_.adjust_access_displacement();
 }
 
+/**** GBMV ****/
+/**
+ * @struct Gbmv
+ * @brief Tree node representing a
+ * vector_ multiplication.
+ */
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_transposed>
+SYCL_BLAS_INLINE
+Gbmv<lhs_t, matrix_t, vector_t, local_range, is_transposed>::Gbmv(
+    lhs_t &_l, matrix_t &_matrix,
+    typename Gbmv<lhs_t, matrix_t, vector_t, local_range,
+                  is_transposed>::index_t &_kl,
+    typename Gbmv<lhs_t, matrix_t, vector_t, local_range,
+                  is_transposed>::index_t &_ku,
+    vector_t &_vector)
+    : lhs_(_l), matrix_(_matrix), vector_(_vector), kl_(_kl), ku_(_ku) {}
+
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_transposed>
+SYCL_BLAS_INLINE typename Gbmv<lhs_t, matrix_t, vector_t, local_range,
+                               is_transposed>::index_t
+Gbmv<lhs_t, matrix_t, vector_t, local_range, is_transposed>::get_size() const {
+  return matrix_.get_size();
+}
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_transposed>
+SYCL_BLAS_INLINE bool
+Gbmv<lhs_t, matrix_t, vector_t, local_range, is_transposed>::valid_thread(
+    cl::sycl::nd_item<1> ndItem) const {
+  return true;
```

Comment:
Maybe add a comment saying that this is handled elsewhere. I got a bit confused (probably due to not being familiar with existing level 2 operations)